sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -0,0 +1,53 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2017 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class AnalysisContextExtensions
+    {
+        public static void CheckReportDiagnostic(this SyntaxNodeAnalysisContext context, Diagnostic diagnostic)

```

Comment:
I guess _ReportDiagnosticIfNotSuppressedInSonarQube(...)_ is a bit long :-)

Alternative suggestions: _CheckAndReportDiagnostic_, _ReportCheckedDiagnostic_ _ReportIfNotSuppressed_?
I don't think any of the alternatives are much of an improvement.

Reply:
I was thinking of `CheckAndReportDiagnostic`, but I did not like the `And` in the name.

Reply:
I quite like _ReportIfNotSuppressed_.

Reply:
I like `ReportCheckedDiagnostic`, also `ReportActiveDiagnostic` `ReportDiagnosticConditional` `ReportDiagnosticIfActive` `ReportDiagnosticWhenActive`... Ping @duncanp-sonar @Evangelink @michalb-sonar